# 🧠 Hypothesis Testing Using *p-value*

---

## 🔹 Step 1: State the Hypotheses

We begin with two hypotheses:

- **Null Hypothesis (H₀):** There is *no effect* or *no difference*.
- **Alternative Hypothesis (H₁ or Hₐ):** There *is an effect* or *a difference*.

For example:

$$
H_0: \mu = \mu_0 \quad \text{and} \quad H_1: \mu \neq \mu_0
$$

---

## 🔹 Step 2: Choose Significance Level (α)

The **significance level (α)** is the probability of rejecting the null hypothesis when it is actually true  
(Type I error).

Common values:

$$
\alpha = 0.05, \; 0.01, \; 0.10
$$

---

## 🔹 Step 3: Compute the Test Statistic

Depending on the type of test (Z-test, T-test, etc.), we calculate a test statistic.

For example, in a **Z-test**:

$$
z = \frac{\bar{X} - \mu_0}{\sigma / \sqrt{n}}
$$

For a **t-test** (when population standard deviation is unknown):

$$
t = \frac{\bar{X} - \mu_0}{s / \sqrt{n}}
$$

---

## 🔹 Step 4: Find the *p-value*

The **p-value** represents the probability of obtaining a test statistic  
as extreme as the observed one, assuming \( H_0 \) is true.

Interpretation:
- A **small p-value** → strong evidence *against* \( H_0 \)
- A **large p-value** → weak evidence *against* \( H_0 \)

---

## 🔹 Step 5: Decision Rule

Compare the p-value with the significance level (α):

$$
\text{If } p \leq \alpha, \; \text{Reject } H_0
$$

$$
\text{If } p > \alpha, \; \text{Fail to Reject } H_0
$$

---

## 💡 Importance of *p-value*

- Quantifies the **strength of evidence** against the null hypothesis.  
- A smaller p-value indicates **stronger evidence** that the null hypothesis may be false.  
- Allows for **objective** decision-making in hypothesis testing.  
- Commonly used in scientific research to determine **statistical significance**.

---

### 📊 Interpretation Table

| p-value | Evidence Against \(H_0\) | Decision (at α = 0.05) |
|:--------:|:--------------------------|:----------------------|
| > 0.05 | Weak evidence | Fail to reject \(H_0\) |
| ≤ 0.05 | Moderate evidence | Reject \(H_0\) |
| ≤ 0.01 | Strong evidence | Reject \(H_0\) confidently |

---

✅ **Summary:**

> The smaller the p-value, the stronger the evidence that the null hypothesis may not be true.

---

![](./img/p-value.jpg)

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
data = pd.read_csv('SOCR-HeightWeight.csv')

In [3]:
data.head()

Index  Height(Inches)  Weight(Pounds)
0      1        65.78331        112.9925
1      2        71.51521        136.4873
2      3        69.39874        153.0269
3      4        68.21660        142.3354
4      5        67.78781        144.2971

In [4]:
pop = data['Weight(Pounds)']

In [7]:
print("Population Mean:", pop.mean())
print("Population Std:",pop.std())

Population Mean: 127.07942116080001
Population Std: 11.660897563604287


In [ ]:
sample = pop.sample(100)

128.643407

In [10]:
print("Population Mean:", sample.mean())
print("Population Std:",sample.std())

Population Mean: 128.643407
Population Std: 10.885129915755707


## 🧩 Hypothesis Setup

We are performing a **one-sample hypothesis test** to determine whether our sample mean is significantly different from the population mean.

### Given Information:
- Sample Mean (x̄) = 128.643407  
- Population Mean (μ₀) = 127.07942116080001  
- Population Standard Deviation (σ) = 11.660897563604287  
- Significance Level (α) = 0.05  

### Objective:
To test whether the sample mean is significantly different from the population mean.

### Hypotheses:

- **Null Hypothesis (H₀):**  
  The sample mean is equal to the population mean.  
  $$
  H_0 : \mu = 127.0794
  $$

- **Alternative Hypothesis (H₁):**  
  The sample mean is not equal to the population mean.  
  $$
  H_1 : \mu \neq 127.0794
  $$

This is a **two-tailed test**, since we are checking for any significant difference (either higher or lower).

---

Next, we will calculate the **test statistic (z)** and the corresponding **p-value** to decide whether to reject or fail to reject the null hypothesis.
